api/envoy/service/ext_proc/v3/external_processor.proto

```diff
@@ -332,3 +344,16 @@ message BodyMutation {
     bool clear_body = 2;
   }
 }
+
+// This message specifies a new timer duration and the original
+// request direction. If these fields are not configured correctly,
+// the response message will be ignored by the Envoy ext_proc filter.
+message NewProcessingTimeout {
+  // Specifies the new timer duration.
+  // Has to be >= 1ms and <= 10000ms.
+  google.protobuf.Duration message_timeout = 1;
+
+  // Specifies the traffic direction of the received request message.
+  // Has to be INBOUND or OUTBOUND.
+  config.core.v3.TrafficDirection traffic_direction = 2;
```

Comment:
nit: Please add a PGV rule to enforce this: https://github.com/bufbuild/protoc-gen-validate#:~:text=State%20x%20%3D%201%20%5B(validate.rules).enum%20%3D%20%7Bnot_in%3A%20%5B1%5D%7D%5D%3B

Reply:
same as above